src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1820,4 +1820,21 @@ else if (tickMap.containsKey(target))
     public void cancelPulse(GlowBlock block) {
         requestPulse(block, 0);
     }
+
+    @Override
+    public int hashCode() {
+        return getUID().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        GlowWorld other = (GlowWorld) obj;
+        return getUID() == other.getUID();
```

Comment:
I agreed just because CraftBukkit used `==`, not that I think CraftBukkit is correct on this. 

I feel like it should use equals because the whole point of equals is to check for object equality, not if they are the same instance. So if the UUIDs have the same bit for bit value, they should be considered equal in `GlowWorld.equals()`. If the same world was loaded twice, it would not return true, as a new UUID would be constructed at https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/blob/master/src/main/java/net/glowstone/io/nbt/NbtWorldMetadataService.java#L41. On the other hand, if we had two of the same world, we wouldn't know for sure if they were equal, block for block.

So with that in mind, another thing to note is plugin compatibility. We must use `==` because that is what CraftBukkit does so that is what plugins will expect.